### ISSUE TITLE
Calling `readf` and `OSCsend` with the same global string causes use after free

### ISSUE THREAD

**Issue by docEdub** [csound/csound#1541]
Calling the `OSCsend` opcode with a global string for an `xdata` arg modified with the `readf` opcode crashes Csound with a use after free.

This .csd crashes on my system:
- Csound 6.16.2
- macOS Big Sur 11.4 

[OSCsend.csd.txt](https://github.com/csound/csound/files/7482142/OSCsend.csd.txt)
```<CsoundSynthesizer>
<CsOptions>
-odac
</CsOptions>
<CsInstruments>

gSBar init "bar"

instr 1
    kI init 0
    gSBar, kI readf "OSCsend.csd"
endin

instr 2
    OSCsend 1, "xenakis.cs.bath.ac.uk", 7770, "/foo/bar", "sis", "FOO", 42, gSBar
endin

</CsInstruments>
<CsScore>
i1 0 2
i2 0 2
</CsScore>
</CsoundSynthesizer>
```

Console output:
```
csound command: Segmentation fault
csound(33020,0x204b03e00) malloc: Incorrect checksum for freed object 0x7fd0114f3f60: probably modified after being freed.
Corrupt value: 0x0
csound(33020,0x204b03e00) malloc: *** set a breakpoint in malloc_error_break to debug
```

Maybe just an issue when `gSBar` gets set to an empty string when `readf` runs out of lines?

**Comment by vlazzarini** [comment]
There are a couple of questions:

1 - does it happen with a local string?
2 - does it happen without using OSCsend? Say using another opcode that just prints the string.

**Comment by vlazzarini** [comment]
I believe it's fixed in git develop now, commit 6e18f0e415c8fe297788952ffb4f8f708323c983

**Comment by docEdub** [comment]
Thank you!

If anybody else runs into this before the next release, it can be worked around by copying the string locally with the `strcpyk` opcode.
